File: app/src/main/java/com/example/myapplication/data/model/SignupModel.java
package com.example.myapplication.data.model;

import com.example.myapplication.utils.Const;
import com.example.myapplication.data.entity.SignupEntity;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;

public class SignupModel {

    public static final int PWD_LENGTH = 4;
    private String filePath;
    private SignupEntity signupEntity;

    public SignupModel() {
        signupEntity = new SignupEntity();
    }

    public SignupEntity getSignupEntity() {
        return signupEntity;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public boolean checkValid() {
        if (Const.isNotNullAndEmpty(
                signupEntity.getUserName(),
                signupEntity.getAge()+"",
                signupEntity.getAddr(),
                signupEntity.getPassword(),
                signupEntity.getRePassword(),
                signupEntity.getGender(),
                filePath) &&
                isLenght4()) {
            return true;
        } else {
            return false;
        }
    }

    public boolean isLenght4() {
        if (getSignupEntity().getPassword().length() == PWD_LENGTH &&
                getSignupEntity().getRePassword().length() == PWD_LENGTH) {
            return true;
        } else {
            return false;
        }
    }


    public boolean checkPwdDuplicate() {
        if (signupEntity.getPassword().equals(signupEntity.getRePassword())) {
            return true;
        } else {
            return false;
        }
    }




    public File getImageFile() {
        File file = new File(filePath);
        return file;
    }

    public MultipartBody.Part getMultipartObj() {
        return Const.prepareFilePart(getImageFile());
    }

    public RequestBody getConvertToRequestBody() {
        return RequestBody.create(MediaType.parse("text/plain"), Const.toJsonString(getSignupEntity()));
    }


    public Map<String,RequestBody> makeParams(){
        Map<String,RequestBody> params = new HashMap<>();
        params.put("nickname",Const.makeRequestBody(signupEntity.getUserName()));
        params.put("password",Const.makeRequestBody(signupEntity.getPassword()));
        RequestBody requestBody = RequestBody.create(MediaType.parse("text/plain"), Const.toJsonString(params));
        return params;
    }




}

File: app_kotlin/src/main/java/com/example/myapplication/data/model/SignupModel.kt
package com.example.myapplication.data.model

import com.example.myapplication.utils.Const
import com.example.myapplication.data.entity.SignupEntity

import java.io.File
import java.util.HashMap

import okhttp3.MediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody

class SignupModel {
    var filePath: String? = null
    val signupEntity: SignupEntity

    val isLenght4: Boolean
        get() = if (signupEntity.password!!.length == PWD_LENGTH && signupEntity.rePassword!!.length == PWD_LENGTH) {
            true
        } else {
            false
        }


    val imageFile: File
        get() = File(filePath!!)

    val multipartObj: MultipartBody.Part
        get() = Const.prepareFilePart(imageFile)

    val convertToRequestBody: RequestBody
        get() = RequestBody.create(MediaType.parse("text/plain"), Const.toJsonString(signupEntity))

    init {
        signupEntity = SignupEntity()
    }

    fun checkValid(): Boolean {
        return Const.isNotNullAndEmpty(
                        signupEntity.userName?:"",
                        signupEntity.age.toString() + "",
                        signupEntity.addr?:"",
                        signupEntity.password?:"",
                        signupEntity.rePassword?:"",
                        signupEntity.gender,
                        filePath?:"") && isLenght4
    }


    fun checkPwdDuplicate(): Boolean {
        return if (signupEntity.password == signupEntity.rePassword) {
            true
        } else {
            false
        }
    }


    fun makeParams(): Map<String, RequestBody> {
        val params = HashMap<String, RequestBody>()
        params["nickname"] = Const.makeRequestBody(signupEntity.userName?:"")
        params["password"] = Const.makeRequestBody(signupEntity.password?:"")
        val requestBody = RequestBody.create(MediaType.parse("text/plain"), Const.toJsonString(params))
        return params
    }

    companion object {

        val PWD_LENGTH = 4
    }


}
